watertap/unit_models/tests/test_gac.py

```diff
@@ -0,0 +1,360 @@
+        )
+        assert (
+            m.fs.unit.config.surface_diffusion_coefficient_type
+            == SurfaceDiffusionCoefficientType.calculated
+        )
+
+        assert m.fs.unit.config.property_package is m.fs.properties
+
+        # only designed for single solute and single solvent (water)
+        assert len(m.fs.unit.config.property_package.solute_set) == 1
+        assert len(m.fs.unit.config.property_package.solvent_set) == 1
+
+    @pytest.mark.unit
+    def test_build_robust(self, gac_frame_robust):
+        m = gac_frame_robust
+
+        # test ports
+        port_lst = ["inlet", "outlet", "adsorbed"]
+        for port_str in port_lst:
+            port = getattr(m.fs.unit, port_str)
+            assert (
+                len(port.vars) == 3
+            )  # number of state variables for NaCl property package
+            assert isinstance(port, Port)
+
+        print(unused_variables_set(m))
+
+        # test statistics
+        assert number_variables(m) == 84
+        assert number_total_constraints(m) == 50
+        assert number_unused_variables(m) == 10  # dens parameters from properties
+
+    @pytest.mark.unit
+    def test_dof_robust(self, gac_frame_robust):
+        m = gac_frame_robust
+        assert degrees_of_freedom(m) == 0
+
+    @pytest.mark.unit
+    def test_calculate_scaling_robust(self, gac_frame_robust):
+        m = gac_frame_robust
+
+        m.fs.properties.set_default_scaling(
+            "flow_mol_phase_comp", 1e-2, index=("Liq", "H2O")
+        )
+        m.fs.properties.set_default_scaling(
+            "flow_mol_phase_comp", 1e5, index=("Liq", "TCE")
+        )
+        calculate_scaling_factors(m)
+
+        # check that all variables have scaling factors
+        unscaled_var_list = list(unscaled_variables_generator(m))
+        assert len(unscaled_var_list) == 0
+
+    @pytest.mark.component
+    def test_initialize_robust(self, gac_frame_robust):
+        initialization_tester(gac_frame_robust)
+
+    @pytest.mark.component
+    def test_var_scaling_robust(self, gac_frame_robust):
+        m = gac_frame_robust
+        badly_scaled_var_lst = list(badly_scaled_var_generator(m))
+        assert badly_scaled_var_lst == []
+
+    @pytest.mark.component
+    def test_solve_robust(self, gac_frame_robust):
+        m = gac_frame_robust
+        results = solver.solve(m)
+
+        # Check for optimal solution
+        assert results.solver.termination_condition == TerminationCondition.optimal
+        assert results.solver.status == SolverStatus.ok
+
+    @pytest.mark.component
+    def test_solution_robust(self, gac_frame_robust):
+        m = gac_frame_robust
+
+        # low tolerance due to inconsistent reporting in reference
+        assert pytest.approx(1.14, rel=1e-1) == value(m.fs.unit.mass_throughput)
+        assert pytest.approx(158 * 3600 * 24, rel=1e-1) == value(m.fs.unit.elap_time)
+        assert pytest.approx(10.68, rel=1e-1) == value(m.fs.unit.bed_area)
```

Comment:
It might be better to use tolerance of 1e-5 and fill in the values you expect to get rather than using the exact reference values. A note about which reference you are referring to might also be useful in the future. Is it the Crittenden reference mentioned earlier?

Reply:
Yes, it is the Crittenden, 2012 reference. It uses calculated values reported in days (e.g., 158 days of operation time) which when converted to seconds for the base SI of the model output introduces a lot of rounding error. I can alter these to the expected value from previous calculations that are the same magnitude but off by rounding to the reference calculations.